add headers to write

lantos1618 opened the pull request:
adds flag to infer and write the headers to csv

lantos1618 committed:
add headers to write

andreaferretti commented on csvtools.nim:
```diff
@@ -351,4 +351,10 @@ proc writeToCsv*[T](ts: openarray[T], path: string, separator = ',',
+      var headers = newSeq[string]()
+      for a,b in fieldPairs(T()):
+        headers.add a
+      f.write(headers.connect())
```
Here you should pass all relevant parameters (separator, escape, etc)

lantos1618 replied:
passed parameters

andreaferretti commented on csvtools.nim:
```diff
@@ -334,7 +334,7 @@ proc writeToCsv*[T](ts: openarray[T], f: var File, separator = ',',
 
 proc writeToCsv*[T](ts: openarray[T], path: string, separator = ',',
-  quote = '\"'; escape = '\"'; quoteAlways = false; dateLayout: string = "") =
+  quote = '\"'; escape = '\"'; quoteAlways = false; dateLayout: string = "", writeHeaders: bool = true) =
```
To avoid changing the default behaviour, I think that `writeHeaders` should default to false

lantos1618 replied:
changed to false

andreaferretti commented:
Hi @LantosHub! Thank you for your contribution! Would you mind fixing the couple of issues I commented on and adding a test?

Have a happy new year!

lantos1618 committed:
add documentation, fix connect option parsing

- add documentation for write headers
- change default writeheaders to false
- parse write options to connect function

LantosHub commented:
Hi added those changes, 
will need to review and ad test cases.

Happy new year!
